src/components/Amount.vue

```diff
@@ -1,25 +1,36 @@
 <template>
     <span class="amount" :class="{ approx: showApprox && isApprox }">
         {{ formattedAmount }}
-        <span class="nim">NIM</span>
+        <span class="currency" :class="currency">{{currency}}</span>
     </span>
 </template>
 
 <script lang="ts">
-import { Component, Prop, Vue } from 'vue-property-decorator';
+import { Component, Prop, Vue, Watch } from 'vue-property-decorator';
+import { formatNumber, moveComma } from '@nimiq/utils';
+import bigInt from 'big-integer';
 
 @Component
 export default class Amount extends Vue {
     // Amount in luna
-    @Prop(Number) public amount!: number;
+    @Prop({type: [Number, Object]}) public amount!: number | bigInt.BigInteger;
     // If set takes precedence over minDecimals and maxDecimals
-    @Prop({type: Number, validator: Amount._validateDecimals }) public decimals?: number;
-    @Prop({type: Number, default: 2, validator: Amount._validateDecimals }) public minDecimals!: number;
-    @Prop({type: Number, default: 5, validator: Amount._validateDecimals }) public maxDecimals!: number;
+    @Prop(Number) public decimals?: number;
+    @Prop({type: Number, default: 2}) public minDecimals!: number;
+    @Prop({type: Number, default: 5}) public maxDecimals!: number;
     @Prop({type: Boolean, default: false}) public showApprox!: boolean;
+    @Prop({type: String, default: 'nim'}) public currency!: string;
+    @Prop({type: Number, default: 5}) public currencyDecimalCount!: number;
 
-    private static _validateDecimals(decimals) {
-        return decimals >= 0 && decimals <= 5 && Number.isInteger(decimals);
+    private approximation = false;
+
+    @Watch('decimals', {immediate: true})
+    @Watch('minDecimals', {immediate: true})
+    @Watch('maxDecimals', {immediate: true})
+    private _validateDecimals(decimals) {
+        if (decimals < 0 || decimals > this.currencyDecimalCount && Number.isInteger(decimals)) {
```

Comment:
This check is not correct.
```suggestion
        if (decimals < 0 || decimals > this.currencyDecimalCount || !Number.isInteger(decimals)) {
```